### ISSUE TITLE
Style configuration changing behavior of savefig

### ISSUE THREAD

**Issue by jasonrobwebster** [matplotlib/matplotlib#9671]
Hi,

I recently upgraded to Matplotlib 2.1.0 and for some reason my fig.savefig functionality has changed. Basically, it now requires a bytes-like object whenever a certain style file is used. Running the following code:

```python
import numpy as np
import matplotlib.pyplot as plt
plt.style.use('aps')

x = np.linspace(0, 1, 100)
y = x**2

plt.plot(x, y)
fig = plt.gcf()

fig.savefig('test.pdf')
```

with the style configuration file 'aps' set to:
```text
font.family	: serif
font.serif	: Times
font.size	: 8.0

text.usetex	: True

xtick.labelsize	: 8
xtick.direction	: in
xtick.major.size: 3

ytick.labelsize	: 8
ytick.direction	: in
ytick.major.size: 3

axes.labelsize	: 8
axes.titlesize	: 8

legend.fontsize	: 7

lines.linewidth	: 1
lines.markersize: 2

figure.figsize	: 3.417, 2.112
figure.dpi	: 100

figure.subplot.left	: 0.133
figure.subplot.right	: 0.99
figure.subplot.top	: 0.97
figure.subplot.bottom	: 0.16

savefig.dpi	: 600
```

I get 

TypeError: a bytes-like object is required, not 'str'

at the line that involves the fig.savefig line. Weirdly, removing plt.style.use('aps') removes this error (but gives me the ugly default style).

Fixing this is simple enough on my side (replace the string with the bytes like obj), but this shouldn't be occurring in fig.savefig - thought I would bring it up.

Important info:
MPL version: 2.1.0
System: Windows 7 Ultimate
Tex backend is MikTex 2.9

Python version:
'3.6.1 (v3.6.1:69c0db5, Mar 21 2017, 18:41:36) [MSC v.1900 64 bit (AMD64)]'

Regards,
jrw

**Comment by jasonrobwebster** [comment]
I jumped the gun, fixing it is not as simple as I thought. Using

```python
fp = open('test.pdf', 'wb')

fig.savefig(fp)

fp.close()
```

executes properly but the pdf does not open. Changing the extension to png does work (and will even allow fig.savefig to accept a string).

**Comment by jklymak** [comment]
Just for fun, remove the `usetex` line...  Not saying you should have to do that, but its the only thing in there that looks like it could cause a problem?

**Comment by jasonrobwebster** [comment]
That fixes it in so far as I can save a pdf file (and I can use ```savefig``` with a ```str``` input). Messes up the point of the style file though.

**Comment by jklymak** [comment]
Can you modify the Issue with the operating system, versions of MPL, python etc, libraries and tex version you are using?  I can't help w/ this (above my pay grade) but its likely necessary info.

FWIW, it works for me, so its something in your setup...

<img width="473" alt="test" src="https://user-images.githubusercontent.com/1562854/32349887-d3625122-bfd5-11e7-8279-7fc8a7922260.png">

**Comment by jklymak** [comment]
Ah, maybe thats it.  I'm working off master...

**Comment by jasonrobwebster** [comment]
Ah okay, so whatever it was seems to be patched up. I'm working off the latest stable build from PyPi. I've downgraded to Matlab 2.0.2 for now, at least til an official stable patch gets released.

**Comment by jklymak** [comment]
Closed by #9292  Thanks for the report @jasonrobwebster and sorry I didn't remember that it'd been fixed earlier...
